### DISCUSSION TITLE
default filter value

### DISCUSSION THREAD

**Discussion by neazk** [Power-Components/livewire-powergrid#1703] [Ideas]
# ⚡ PowerGrid ⚡ Feature Request

## Summary

default filter value

### Why is this needed?

example: to hide disabled users by default 

### Suggested Solution(s)

a new method default() or defaultValue() to set default value

```
Filter::boolean('is_active', 'is_active')
                ->default(true),
```

**Comment by nguyenhoang24695** [comment]
Same issue

**Comment by LoLei** [comment]
I would also like to have the option to persist filter values across page reloads / page switches. Is that possible somehow without extensive manual implementations?

Update: I did just find this, haven't tested it yet: https://livewire-powergrid.com/table-component/component-configuration.html#persist

**Comment by johannesMEGABAD** [comment]
Bump so that it maybe gets implemented

**Comment by DevTerabytio** [comment]
I create this function to do these, but I don't know is this correct form to-do that
```php
public function setDefaultFilter(string $field, string $label, mixed $value): void
    {
        $filters = $this->filters();
        $filter = collect($filters)->firstWhere('field', "===",$field);
        if (!isset($filter)) {
            return;
        }
        data_set($this->filters, "{$filter->key}.$field", $value);
        $this->addEnabledFilters($field, $label);
    }
```

**Reply by DevTerabytio** [reply]
While this implemented to the PG, you can create a trait for your tables and used, called in setUP function NOT USE IN FILTERS()

**Reply by johannesMEGABAD** [reply]
```php
public function setDefaultFilter(string $field, string $label, mixed $value): void
{
    /** @var object|null $filter */
    $filter = collect($this->filters())
        ->first(fn ($filter) => $filter->field === $field);

    if ($filter === null) {
        return;
    }

    data_set($this->filters, "{$filter->key}.{$field}", $value);

    $this->addEnabledFilters($field, $label);
}
```

This version uses a strict PHP comparison on the filter object instead of relying on Laravel’s **firstWhere** operator handling. Maybe it can be implemented somehow into PowerGrid. Right now I use it inside a Trait that I then make use of in my tables. Thanks @neazk for the idea!

**Comment by luanfreitasdev** [comment]
Feel free to open a PR for this.

**Reply by johannesMEGABAD** [reply]
https://github.com/Power-Components/livewire-powergrid/pull/2054

I opened a PR for that feature :)
